src/pages/chat/index.astro

```diff
@@ -2,9 +2,17 @@
 import { Party } from "../../components/Party";
 import Layout from "../../layouts/Layout.astro";
 
-const username = Astro.url.searchParams.get("username") ?? "anonymous";
+const usernameParams = Astro.url.searchParams.get("username");
+const username =
+  usernameParams === null || usernameParams === ""
+    ? "anonymous"
+    : usernameParams;
 
-const roomId = Astro.url.searchParams.get("roomId") ?? "astro-partykit-room";
+const roomIdParams = Astro.url.searchParams.get("roomId");
+const roomId =
+  roomIdParams === null || roomIdParams === ""
+    ? "astro-partykit-room"
+    : roomIdParams;
```

Comment:
nit: naming of `usernameParams`. I typically user `rawUsername`.
 
refactor (non-blocking): This works perfectly for checking for null or an empty string. It can be more terse.

```suggestion
const usernameParams = Astro.url.searchParams.get("username");
const username = !usernameParams ? "anonymous" : usernameParams;

const roomIdParams = Astro.url.searchParams.get("roomId");
const roomId = !roomIdParams ? "astro-partykit-room" : roomIdParams;
```